pkg/datapath/iptables/iptables.go

```diff
@@ -828,6 +829,160 @@ func (m *IptablesManager) installForwardChainRules(ifName, localDeliveryInterfac
 	return nil
 }
 
+func (m *IptablesManager) installMasqueradeRules(prog, ifName, localDeliveryInterface,
```

Comment:
Just a note that it would have been easier to review if you created the method in a separate commit.

Reply:
Sorry if I misunderstood, but are you suggesting that in the first commit I add these rules wrapped inside an `if option.Config.EnableIPv6 { ... }` inside `InstallRules` and then in separate commit merge rules for IPv4 and IPv6 using a single function?

Reply:
I meant in the first commit to move the existing IPv4 code into this method, and then in the second commit to add the IPv6 handling.

Reply:
Makes sense, Done!